dokka-runners/dokka-gradle-plugin/src/main/kotlin/DokkaPlugin.kt

```diff
@@ -23,25 +22,26 @@ abstract class DokkaPlugin
 @DokkaInternalApi
 constructor() : Plugin<Project> {
 
-    override fun apply(target: Project) {
-        val dokkaMode = target.dokkaMode()
-        if (dokkaMode.dokkatooEnabled) {
-            if (!dokkaMode.dokkatooEnabledQuietly) {
-                logger.lifecycle("[${target.displayName}] Dokka Gradle Plugin: Dokkatoo mode enabled")
-            }
-            with(target.pluginManager) {
-                apply(type = DokkaBasePlugin::class)
-
-                // auto-apply the custom format plugins
-                apply(type = DokkaHtmlPlugin::class)
-            }
+    override fun apply(project: Project) {
+        val pluginFeaturesService = project.pluginFeaturesService
+        if (pluginFeaturesService.enableV2Plugin) {
+            applyV2(project)
         } else {
-            logger.warn("[${target.path}] Dokka Gradle Plugin: classic mode, please migrate, see https://kotl.in/dokka-gradle-migration")
-            target.pluginManager.apply(ClassicDokkaPlugin::class)
+            applyV1(project)
         }
     }
 
-    companion object {
-        private val logger = Logging.getLogger(DokkaPlugin::class.java)
+    private fun applyV1(project: Project) {
+        project.pluginManager.apply(ClassicDokkaPlugin::class)
     }
+
+    private fun applyV2(project: Project) {
+        with(project.pluginManager) {
+            apply(type = DokkaBasePlugin::class)
+            // auto-apply the HTML plugin
+            apply(type = DokkaHtmlPlugin::class)
+        }
+    }
+
+    companion object
```

Comment:
I intentionally kept the companion object in case someone wants to create an extension on DokkaPlugin. If it's not defined, then it completely prevents them, which can be very annoying as a user.